eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -80,10 +80,15 @@ jobs:
         inputs:
           ignoreDirectories: "sdk/storage/Azure.Storage.Common/swagger/Generator"
         condition: and(succeededOrFailed(), ne(variables['Build.Reason'],'PullRequest'))
+  - job: VerifyGenerated
```

Comment:
Is there a reason why this can't be run within the Analyze job (on Windows)? Adding this job here causes a whole agent to be consumed.

Reply:
Analyze runs after build completes and we would like to start the code check as early as possible because it's the longest leg of the build. In my initial change, I removed the Analyze job and moved all the steps to build to reduce the agent found but @weshaggard wanted to preserve it.

Reply:
@mitchdenny This is a pretty long running task which would be good to run in parallel with the build and test legs. I believe the goal was to have the Analyze job run after the build job which makes the overall pipeline run take longer then it should. The original reason was because we wanted to download and analyze the packages from the build job, however it looks like we aren't even downloading the packages any more and as such the [Component Governance](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=536064&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=72a3110e-44c0-5323-690d-8dd09cc84ac8) is not running on any packages beyond what is committed in the repo (yes there are still a couple of those :(). I just looked at our [CG runs](https://dev.azure.com/azure-sdk/internal/_componentGovernance/98522?_a=components&typeId=1937810&alerts-view-option=active) and the packages are present and it looks like they are getting analyzed in our [dev publishing step](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=536064&view=logs&j=b1e79959-24d8-5aa9-2799-72d40c3e051b&t=f3fc8d9f-4e68-567c-1c63-cd6d9a749ff5) instead.  

Given that @pakrym can you just keep this in the Analyze job and just remove the Depends on Build from that job. 

I've filed https://github.com/Azure/azure-sdk-for-net/issues/15143 to follow-up on the CG stuff.

Reply:
>Given that @pakrym can you just keep this in the Analyze job and just remove the Depends on Build from that job.

CodeGeneration is significantly faster on ubuntu, that's part of the speedup.

Reply:
I'm fine with the move to ubuntu as well.